pkg/shp/cmd/build/run.go

```diff
@@ -86,44 +87,73 @@ func (c *RunCommand) Run(params *params.Params, ioStreams *genericclioptions.IOS
 		return nil
 	}
 
-	err := wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
-		if err := brr.Get(c.cmd.Context(), buildRun.Name, buildRun); err != nil {
-			fmt.Fprintf(ioStreams.ErrOut, "BuildRun %q not available: %q\n", buildRun.Name, err.Error())
-			return false, nil
-		}
-
-		if buildRun.Status.LatestTaskRunRef == nil {
-			fmt.Fprintf(ioStreams.ErrOut, "BuildRun %q created but cannot follow logs since there is no TaskRun reference\n", buildRun.Name)
-			return false, nil
-		}
-		return true, nil
-	})
+	watch, err := brr.Watch(c.cmd.Context(), v1.ListOptions{})
 	if err != nil {
 		return err
 	}
+	taskRunName := ""
+	for len(taskRunName) == 0 {
+		select {
```

Comment:
I did take a stab at trying to generalize this and create a generic watch func in https://github.com/shipwright-io/cli/blob/main/pkg/shp/util/crud.go but it got klunky pretty quick @otaviof 

Curious on your thoughts here.

Reply:
Yes! It looks a whole lot better! Thanks.